modules/light-clients/08-wasm/types/errors.go

```diff
@@ -19,4 +19,5 @@ var (
 	ErrWasmContractCallFailed          = errorsmod.Register(ModuleName, 14, "wasm contract call failed")
 	ErrWasmInvalidResponseData         = errorsmod.Register(ModuleName, 15, "wasm contract returned invalid response data")
 	ErrWasmInvalidContractModification = errorsmod.Register(ModuleName, 16, "wasm contract made invalid state modifications")
+	ErrVMError                         = errorsmod.Register(ModuleName, 17, "wasm VM error")
```

Comment:
would the intension to be to use in this in favour of `ErrWasmContractCallFailed` in all places? 

Is there a difference between the two errors and when one should be used in favour of the other?

Reply:
Not in all places. The way I've done it is like this:

```
res, err := <contract call>
if err != nil {
  return result, errorsmod.Wrap(ErrVMError, err.Error())
}
if res.Err != "" {
  return result, errorsmod.Wrap(ErrWasmContractCallFailed, res.Err)
}
```

Reply:
Aha! Perfect, that makes sense!